src/v/cloud_storage/remote_partition.cc

```diff
@@ -69,6 +71,134 @@ get_kafka_max_offset(const partition_manifest::segment_meta& m) {
     return m.committed_offset - delta;
 }
 
+remote_partition::iterator
+remote_partition::materialize_segment(const segment_meta& meta) {
+    auto key_offset = meta.base_offset - meta.delta_offset;
```

Comment:
nit: not really key_offset anymore as the key in the `_segments` map is the redpanda offset, right?

Reply:
fixed